[1.16] Adds workflow limitations (#4865)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-dynamodb.md

}
```

## Workflow Limitations

> The more complex a workflow is (number of activities, child workflows, etc.), the more state operations it will perform per state store transaction.

Reviewer: Suggesting adding here that DynamoDB is not recommended for production when using workflow. WDYT?

> The more complex a workflow is (number of activities, child workflows, etc.), the more state operations it will perform per state store transaction.

Reviewer: ```suggestion
The more complex a workflow is (number of activities, child workflows, etc.), the more state operations it performs per state store transaction.
```

The maximum number of operations that can be performed by DynamoDB in a [single transaction is 100](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/transaction-apis.html).

> This means that DynamoDB can only hadle workflows with a limited complexity, meaning it is not suitable for all workflow scenarios.

Reviewer: ```suggestion
This means that DynamoDB can only handle workflows with a limited complexity, meaning it is not suitable for all workflow scenarios.
```

}
```

## Workflow Limitations

{{% alert title="Note" color="primary" %}}

As described below, DynamoDB has limitations that likely make it unsuitable for production environments.

> There is currently no path for migrating Workflow data from DynamoDB to another state store, meaning exceeding these limits in production will result in failed workflows this no workaround.

Reviewer: ```suggestion
There is currently no path for migrating Workflow data from DynamoDB to another state store, meaning exceeding these limits in production will result in failed workflows with no workaround.
```
